### DISCUSSION TITLE
Pylance not finding installed editable package

### DISCUSSION THREAD

**Discussion by dvoraen** [microsoft/pylance-release#3975] [Q&A] [answered]
Have a question about unresolved imports, and this may be something on my end I haven't configured right because I did a system restore recently.

Before I had to reset my rig, I had two cloned repositories, A and B.  A depends on B; it's an extension of B.  B was installed in editable mode inside A's venv and shows up in `pip list`.  A and B I set up as a multi-root workspace inside VSCode in case I needed to refer to B's source.  I can't recall doing anything extra, but Pylance would find imports referencing B just fine.

This is the directory setup:

```
OuterFolder
- A
  - venv (inside A) with B installed in editable mode
- B

workspace file in another drive (for automatic backup reasons)
```

After I reset my rig, I have what I think is the same setup, only now Pylance is stating it can't find imports involving B unless I put B in `python.analysis.extraPaths`.  I am certain I did not have this set before the reset.

So my question is: why is it necessary?  Am I forgetting something?  Did something change and I don't know it?  I could swear Pylance would check for installed packages for imports, even ones in editable mode, but it doesn't seem able to find this one.

Thank you in advance.

**Comment by erictraut** [comment, marked as answer]
It's likely related to changes in the way editable installs work. See the [troubleshooting docs](https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#editable-install-modules-not-found) for details.

**Reply by dvoraen** [reply]
Yep, that was it.  Reinstalling it via compat mode (as described by setuptools) has things as they used to be.

Thank you, Eric!
